vowpalwabbit/parser/flatbuffer/parse_example_flatbuffer.cc

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include <fstream>
+#include <iostream>
+#include <cfloat>
+
+#include "../../global_data.h"
+#include "../../constant.h"
+#include "../../cb.h"
+#include "../../action_score.h"
+#include "../../best_constant.h"
+#include "parse_example_flatbuffer.h"
+
+namespace VW
+{
+namespace parsers
+{
+namespace flatbuffer
+{
+int flatbuffer_to_examples(vw* all, v_array<example*>& examples)
+{
+  return static_cast<int>(all->flat_converter->parse_examples(all, examples));
+}
+
+parser::parser(std::string filename) : _filename(filename), _flatbuffer_pointer(nullptr), _example_index(0), _c_hash(0)
+{
+}
+
+parser::parser(uint8_t* buffer_pointer)
+    : _filename("empty"), _flatbuffer_pointer(buffer_pointer), _example_index(0), _c_hash(0)
+{
+  _data = VW::parsers::flatbuffer::GetExampleCollection(_flatbuffer_pointer);  // Shift as a separate call
+}
+
+void parser::init()
+{
```

Comment:
Why make init separate? Not it is possible to call parse_examples without it being in a valid state?

Reply:
The way we parse examples will change a bit. The work has already been done along with all the benchmarking work and might be better if it comes in the codebase in smaller PRs to make reviewing easier. What do you think?

Reply:
If it's addressed in upcoming PRs I'm happy